### PR TITLE
tests: Clean up unnecessary try-catch blocks

### LINKED ISSUE #6644: Don't catch exceptions in Java unit tests
In many of our unit tests, we have things like that:

https://github.com/OpenRefine/OpenRefine/blob/62a732226fddf529caea0aa85ed8a635541d6626/main/tests/server/src/com/google/refine/exporters/sql/SqlExporterTests.java#L125-L129

This is a bad practice because it hides the exception's cause and message by throwing an unrelated exception.

Instead, test methods should have any required `throws` metadata in the signature so that this exception catching can be omitted. The test framework is capable of catching such exceptions and reporting the corresponding test as having an error.

More reading: https://softwareengineering.stackexchange.com/questions/101952/should-i-use-try-catch-in-my-test-methods

### PR DESCRIPTION
**Pull request by wetneb**
Closes #6644.

### CODE REVIEW

**Comment by tfmorris** [comment]
Was this done with the help of an IDE's refactoring tools? Trying to figure out how much attention I need to pay to the voluminous changes...

**Comment by wetneb** [comment]
No, I had to do most of it manually. If you know of a technique to automate something like this, I am interested. I thought about recipes from OpenRewrite but they don't seem to have any (the closest I could find was [normalizing the `throws` statements in tests](https://docs.openrewrite.org/recipes/java/testing/cleanup/simplifytestthrows)) and I didn't feel like writing one (they make it rather hard to run their tool locally to push people to their cloud service, which discourages me from investing in it).

If you find it a good use of your time to scrutinize all of it, I would recommend doing it with a diff tool like [difftastic](https://github.com/Wilfred/difftastic) because it gives a much clearer picture.

File: main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java [hunk 1]
```diff
@@ -623,41 +551,27 @@ public void processRecordsFromComplexXmlStructure() throws IOException {
...
     }
 
-    public void loadData(String xml) {
-        try {
-            inputStream = new ByteArrayInputStream(xml.getBytes("UTF-8"));
-        } catch (UnsupportedEncodingException e1) {
-            Assert.fail();
-        }
+    public void loadData(String xml) throws Exception {
+        inputStream = new ByteArrayInputStream(xml.getBytes("UTF-8"));
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java:563]:
```suggestion
        inputStream = new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8));
```

File: main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java [hunk 2]
```diff
@@ -623,41 +551,27 @@ public void processRecordsFromComplexXmlStructure() throws IOException {
...
         loadData(JsonImporterTests.getSample());
     }
 
-    public void loadData(String xml) {
-        try {
-            inputStream = new ByteArrayInputStream(xml.getBytes("UTF-8"));
-        } catch (UnsupportedEncodingException e1) {
-            Assert.fail();
-        }
+    public void loadData(String xml) throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java:562]:
```suggestion
    public void loadData(String xml) {
```

File: main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java [hunk 3]
```diff
@@ -623,41 +551,27 @@ public void processRecordsFromComplexXmlStructure() throws IOException {
...
     }
 
-    public void ParserSkip() {
-        try {
-            if (parser.current() == TreeReader.Token.Ignorable) {
-                parser.next(); // move parser forward once e.g. skip the START_DOCUMENT parser event
-            }
-        } catch (Exception e1) {
-            Assert.fail();
+    public void ParserSkip() throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java:566]: I think for these low level helpers where they throw one or two  very specific exceptions, it's probably better to declare the actual exceptions thrown

```suggestion
    public void ParserSkip() throws TreeReaderException {
```

File: main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java [hunk 4]
```diff
@@ -623,41 +551,27 @@ public void processRecordsFromComplexXmlStructure() throws IOException {
...
-    public TreeReader createXmlParser() {
-        try {
-            parser = new XmlParser(inputStream);
-            return parser;
-        } catch (XMLStreamException e) {
-            return null;
-        } catch (IOException e) {
-            return null;
-        }
+    public TreeReader createXmlParser() throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java:572]:
```suggestion
    public TreeReader createXmlParser() throws IOException, XMLStreamException  {
```

File: main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java [hunk 5]
```diff
@@ -623,41 +551,27 @@ public void processRecordsFromComplexXmlStructure() throws IOException {
     }
 
     // ----------------helpers-------------
-    public void loadSampleXml() {
+    public void loadSampleXml() throws Exception {
         loadData(XmlImporterTests.getSample());
     }
 
-    public void loadSampleJson() {
+    public void loadSampleJson() throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java:558]: This method appears to be unused, but it doesn't throw any exceptions.

```suggestion
    public void loadSampleJson() {
```

File: main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java [hunk 6]
```diff
@@ -623,41 +551,27 @@ public void processRecordsFromComplexXmlStructure() throws IOException {
     }
 
     // ----------------helpers-------------
-    public void loadSampleXml() {
+    public void loadSampleXml() throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImportUtilitiesTests.java:554]:
```suggestion
    public void loadSampleXml() {
```

Doesn't throw with the charset fix

File: main/tests/server/src/com/google/refine/importers/XmlImporterTests.java [hunk 7]
```diff
@@ -422,23 +421,14 @@ public static String getSampleWithTreeStructure() {
...
     }
 
-    private void RunTest(String testString, ObjectNode objectNode) {
-        try {
-            inputStream = new ByteArrayInputStream(testString.getBytes("UTF-8"));
-        } catch (UnsupportedEncodingException e1) {
-            Assert.fail();
-        }
+    private void RunTest(String testString, ObjectNode objectNode) throws Exception {
+        inputStream = new ByteArrayInputStream(testString.getBytes("UTF-8"));
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImporterTests.java:429]:
```suggestion
        inputStream = new ByteArrayInputStream(testString.getBytes(StandardCharsets.UTF_8));
```
Same pattern as JSON importer. Fix this and then undo all the added `throws` clauses up the call stack

File: main/tests/server/src/com/google/refine/importers/XmlImporterTests.java [hunk 8]
```diff
@@ -422,23 +421,14 @@ public static String getSampleWithTreeStructure() {
...
         RunTest(testString, getOptions(job, SUT));
     }
 
-    private void RunTest(String testString, ObjectNode objectNode) {
-        try {
-            inputStream = new ByteArrayInputStream(testString.getBytes("UTF-8"));
-        } catch (UnsupportedEncodingException e1) {
-            Assert.fail();
-        }
+    private void RunTest(String testString, ObjectNode objectNode) throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImporterTests.java:428]:
```suggestion
    private void RunTest(String testString, ObjectNode objectNode) {
```

File: main/tests/server/src/com/google/refine/importers/XmlImporterTests.java [hunk 9]
```diff
@@ -422,23 +421,14 @@ public static String getSampleWithTreeStructure() {
         return sb.toString();
     }
 
-    private void RunTest(String testString) {
+    private void RunTest(String testString) throws Exception {
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/importers/XmlImporterTests.java:424]:
```suggestion
    private void RunTest(String testString) {
```

File: main/tests/server/src/com/google/refine/grel/GrelTests.java [hunk 10]
```diff
@@ -80,30 +81,20 @@ public void testInvalidSyntax() {
...
         };
         for (String test : tests) {
-            try {
-                MetaParser.parse("grel:" + test);
-            } catch (ParsingException e) {
-                // Test succeeded
-                continue;
-            }
-            fail("Expression failed to generate parse syntax error: " + test);
+            assertThrows(ParsingException.class, () -> MetaParser.parse("grel:" + test));
```
Comment (tfmorris) [inline main/tests/server/src/com/google/refine/grel/GrelTests.java:84]:
```suggestion
            assertThrows("Expression failed to generate parse syntax error: " + test, ParsingException.class, () -> MetaParser.parse("grel:" + test));
```
For loop based tests, it's valuable to know which iteration failed

**Review by tfmorris — CHANGES_REQUESTED** [review]
Thanks for all the test cleanup. That was a lot of work!

My eyes were kind of glazing over by the end, but hopefully I caught the bulk of the issues. All in all things look good with just a few minor improvements needed. It'll be a nice simplification of the tests.

File: main/tests/server/src/com/google/refine/commands/util/CancelProcessesCommandTests.java [hunk 11]
```diff
@@ -196,56 +156,36 @@ public void doPostRegressionTest() {
...
-        } catch (IOException e1) {
-            Assert.fail();
-        }
+        when(response.getWriter()).thenReturn(pw);
 
         // run
-        try {
-            SUT.doPost(request, response);
-        } catch (ServletException e) {
-            // expected
```
[the remark this answers is not in the document]
Comment (wetneb) [reply main/tests/server/src/com/google/refine/commands/util/CancelProcessesCommandTests.java:215]: Indeed, it is caught and converted to an HTTP response with `{"code": "error", …}`, so I have updated the test to reflect that

**Review by tfmorris — APPROVED** [review]

**Pull request merged**